app/Libraries/Search/BeatmapsetSearchRequestParams.php

```diff
@@ -68,15 +68,15 @@ public function __construct(Request $request, ?User $user = null)
         }
 
         $sort = explode('_', $request['sort']);
-        $this->sort = static::normalizeSort(
+        $this->sort = $this->normalizeSort(
```

Comment:
Not related with this pull request but this assignment and related functions seem a bit... interesting.

- first it create sort with given params
- and then the created sort is thrown out, replaced with another pretty much the same sort just the `sort[0]` mapped to correct field
- it's put in array
- the `normalizeSort` function loops the array which is always single element.
- (kind of related to this pull request) goes to great length using `array_unshift` instead of just doing it before appending with provided sort
  - it prepends the resulting sort, potentially overriding any previous sorts (`['artist', 'approved_date']` &rarr; `['queued_at', 'artist', 'approved_date']`)
  - except it doesn't matter because there's always only one sort

Reply:
probably should have added a comment there >_>